frontend/src/d3/GraphSimulation.ts

```diff
@@ -44,28 +50,60 @@ export default class {
   private width: number;
   private height: number;
   private nodes: Array<GraphNode>;
+  private unfilteredNodes: Array<GraphNode>;
   private edges: Array<D3Edge | GraphEdge>;
+  private unfilteredEdges: Array<D3Edge | GraphEdge>;
   private scale: number = 1;
+  private nodeFilter: GraphNodeFilter;
 
   constructor(
     svg: SVGSVGElement,
     width: number,
     height: number,
-    initialNode: Node,
+    initialNode: GraphNode,
     onClickNode: (node: GraphNode) => void,
+    nodeFilter: GraphNodeFilter,
   ) {
     this.svg = d3.select(svg);
     this.edgeSvg = this.svg.append('g');
     this.nodeSvg = this.svg.append('g');
     this.width = width;
     this.height = height;
-    this.nodes = [initialNode as GraphNode];
+    this.nodes = [initialNode];
+    this.unfilteredNodes = [initialNode];
     this.edges = [];
+    this.unfilteredEdges = [];
     this.onClickNode = onClickNode;
+    this.nodeFilter = nodeFilter;
     this.initZoom();
     this.forceSimulation = this.initForceSimulation();
   }
 
+  removeDisconnectedEdges = () => {
+    this.edges = this.unfilteredEdges.filter((edge) => {
+      const source = typeof edge.source === 'string' ? edge.source : edge.source.id;
+      const target = typeof edge.target === 'string' ? edge.target : edge.target.id;
+      return (
+        this.nodes.some((node) => node.id === source) &&
+        this.nodes.some((node) => node.id === target)
+      );
+    });
+  };
+
+  applyFilter = () => {
+    // const toggle = this.nodes.length === this.unfilteredNodes.length;
```

Comment:
Fjern unødvendige kommentarer!

Reply:
Remember that all communication on GitHub should be in English :))